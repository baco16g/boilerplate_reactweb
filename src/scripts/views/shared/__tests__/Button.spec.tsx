import * as React from 'react';
import {render, cleanup, fireEvent} from 'react-testing-library';
import 'jest-dom/extend-expect';
import 'jest-styled-components';
import Button from '../Button';

afterEach(cleanup);

it('matches snapshot', () => {
  const onPress = jest.fn();
  const {asFragment} = render(<Button label="Press!!" onPress={onPress} />);
  expect(asFragment()).toMatchSnapshot();
});

it('The event should fires when the button is clicked', () => {
  const onPress = jest.fn();
  const {getByText} = render(<Button label="Press!!" onPress={onPress} />);
  fireEvent.click(getByText('Press!!'));
  expect(onPress).toHaveBeenCalled();
});

it('The style passed in props should be applied correctly', () => {
  const onPress = jest.fn();
  const {container} = render(
    <Button label="Press!!" color="#ccc" fontsize={12} onPress={onPress} />
  );
  expect(container.firstChild).toHaveStyleRule('color', '#ccc');
  expect(container.firstChild).toHaveStyleRule('border', 'solid 2px #ccc');
  expect(container.firstChild).toHaveStyleRule('background', '#ccc', {modifier: ':hover'});
  expect(container.firstChild).toHaveStyleRule('font-size', '12px');
});
