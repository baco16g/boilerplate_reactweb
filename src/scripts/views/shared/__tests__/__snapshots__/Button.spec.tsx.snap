// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  .c0 {
  display: inline-block;
  padding: 0.3em 1em;
  -webkit-text-decoration: none;
  text-decoration: none;
  font-size: 12px;
  color: #67c5ff;
  border: solid 2px #67c5ff;
  border-radius: 3px;
  -webkit-transition: .4s;
  transition: .4s;
}

.c0:hover {
  background: #67c5ff;
  color: white;
}

<button
    class="c0"
    color="#67c5ff"
  >
    Press!!
  </button>
</DocumentFragment>
`;
